src/main/java/racingcar/RacingCar.java

```diff
@@ -1,63 +1,33 @@
 package racingcar;
 
 import racingcar.domain.Car;
+import racingcar.domain.Cars;
+import racingcar.domain.Race;
+import racingcar.generator.RandomGenerator;
+import racingcar.strategy.RandomStrategy;
 import racingcar.ui.InputView;
-import racingcar.ui.ResultView;
 
-import java.util.Arrays;
-import java.util.Random;
+import java.util.ArrayList;
+import java.util.List;
 
 public class RacingCar {
 
-    public static final Random RANDOM = new Random();
-    public static final int RANDOM_NUMBER = 4;
-    public static final int RANDOM_MAX_NUMBER = 10;
-    public static final int CAR_MIN = 0;
-    public static final int NUM_MIN = 0;
-
-    public static void racingCar() {
-        int car = InputView.inputNumberOfCar();
-        int num = InputView.inputNumberOfAttempts();
-
-        validateInput(car, num);
-        racing(num, getCars(car));
-    }
-
-    public static void validateInput(int car, int num) {
-        if (car <= CAR_MIN || num <= NUM_MIN) {
-            throw new IllegalArgumentException("차량 수와 시도 횟수는 0보다 큰 수만 입력 가능합니다.");
-        }
-    }
-
-    private static void racing(int num, Car[] cars) {
-        System.out.println("실행 결과");
-        for (int i = 0; i < num; i++) {
-            moveCar(cars);
-            ResultView.viewRacingCar(cars);
-        }
-    }
-
-    public static void moveCar(Car[] cars) {
-        for (int j = 0; j < cars.length; j++) {
-            if (cars[j].getMoveCount() == 0) {
-                cars[j].reset();
-                continue;
-            }
-            if (getRandomNumber() >= RacingCar.RANDOM_NUMBER) {
-                cars[j].move();
-            }
-        }
+    public static void gameStart() {
+        Cars cars = new Cars(getCars(InputView.inputValidatedNumberOfCar()));
+        Race race = new Race(cars, InputView.inputValidatedNumberOfAttempts());
+        race.start(createRandomStrategy());
     }
 
-    public static Car[] getCars(int car) {
-        Car[] cars = new Car[car];
-        for (int i = 0; i < car; i++) {
-            cars[i] = new Car();
+    public static ArrayList<Car> getCars(int car) {
+        ArrayList<Car> cars = new ArrayList<>();
```

Comment:
소소한 의견입니다만 다형성을 위해서라도 `List<Car>`사용을 권장 드립니다 🙏 
```suggestion
        List<Car> cars = new ArrayList<>();
```